logilica_weekly_report/page_settings.py

```diff
@@ -0,0 +1,311 @@
+              repositories:
+                repositories to be added
+              failures:
+                failures object, will be updated in case of failures
+            """
+
+            added_repositories = []
+            for repo_slug in wait_for_available_repositories(repositories):
+                if not self.has_repository_imported(repo_slug):
+                    if self.add_membership_repository(repo_slug):
+                        added_repositories.append(repo_slug)
+                    else:
+                        failures[(connector, integration_name)].append(
+                            (
+                                repo_slug,
+                                f"❌ Repository {repo_slug} was not among available repositories",
+                            )
+                        )
+            missing_repositories = self.check_imported_repositories(added_repositories)
+            if missing_repositories:
+                failures[(connector, integration_name)].extend(missing_repositories)
+
+        sync_failures: IntegrationSyncFailures = defaultdict(list)
+        for integration_name, details in integrations.items():
+            connector = details["connector"]
+            public_repos = details.get("public_repositories", [])
+            member_repos = details.get("membership_repositories", [])
+            logging.debug(
+                "Syncing integration '%s' (connector '%s'), "
+                "%d public repositories, %d membership based repositories",
+                integration_name,
+                connector,
+                len(public_repos),
+                len(member_repos),
+            )
+
+            # open configuration UI
+            self.open_settings("Integrations")
+            self.open_integration_configuration(
+                integration=integration_name, connector=connector
+            )
+
+            # process public repositories
+            process_public_repositories(
+                connector=connector,
+                integration_name=integration_name,
+                repositories=public_repos,
+                failures=sync_failures,
+            )
+            # process membership repositories
+            process_membership_repositories(
+                connector=connector,
+                integration_name=integration_name,
+                repositories=member_repos,
+                failures=sync_failures,
+            )
+
+        if sync_failures:
+            logging.error(
+                "There are failures for %d integrations in syncing sources",
+                len(sync_failures),
+            )
+            for (connector, name), failures in sync_failures.items():
+                logging.error("Failures in connector %s named '%s':", connector, name)
+                for repo, f in failures:
+                    logging.error("%s: %s", repo, f)
+                raise RuntimeError(sync_failures)
```

Comment:
I believe that the `raise` statement is indented too far:  it should be after the `for` loop at line 200, not inside it.

Reply:
good catch.